connect/api/src/main/java/org/apache/kafka/connect/sink/SinkRecord.java

```diff
@@ -20,18 +20,25 @@
 import org.apache.kafka.connect.connector.ConnectRecord;
 import org.apache.kafka.connect.data.Schema;
 import org.apache.kafka.connect.header.Header;
+import org.apache.kafka.connect.transforms.Transformation;
+
+import java.util.Map;
+import java.util.Objects;
 
 /**
- * SinkRecord is a {@link ConnectRecord} that has been read from Kafka and includes the kafkaOffset of
- * the record in the Kafka topic-partition in addition to the standard fields. This information
- * should be used by the {@link SinkTask} to coordinate kafkaOffset commits.
+ * SinkRecord is a {@link ConnectRecord} that has been read from Kafka and includes the original Kafka record's
+ * topic, partition and offset (before any {@link Transformation}s have been applied) in addition to the standard fields.
+ * This information should be used by the {@link SinkTask} to coordinate offset commits.
```

Comment:
Nit: formatting/readability
```suggestion
 * topic, partition and offset (before any {@link Transformation transformations}
 * have been applied) in addition to the standard fields. This information should be used by the {@link SinkTask} to coordinate
 * offset commits.
```